pkg/sds/server/reconciler.go

```diff
@@ -76,22 +80,37 @@ func (d *DataplaneReconciler) Reconcile(dataplaneId core_model.ResourceKey) erro
 	return nil
 }
 
-func (d *DataplaneReconciler) shouldGenerateSnapshot(proxyID string, mesh *mesh_core.MeshResource) (bool, error) {
+func (d *DataplaneReconciler) shouldGenerateSnapshot(proxyID string, mesh *mesh_core.MeshResource) (bool, string, error) {
 	currentSnapshot, err := d.cache.GetSnapshot(proxyID)
-	if err != nil { // snapshot does not exist
-		return true, nil
+	if err != nil {
+		return true, "Snapshot does not exist", nil
 	}
 
-	parts := strings.Split(currentSnapshot.GetVersion(envoy_resource.SecretType), "-")
+	parts := strings.Split(currentSnapshot.GetVersion(envoy_resource.SecretType), ";")
 	if len(parts) != 2 {
-		return false, errors.New(`invalid snapshot version format. Format should be "UnixNano-NameOfTheCA"`)
+		return false, "", errors.New(`invalid snapshot version format. Format should be "UnixNano-NameOfTheCA"`)
 	}
 	// generate snapshot if CA changed
 	caName := parts[1]
 	if caName != mesh.GetEnabledCertificateAuthorityBackend().Name {
-		return true, nil
+		return true, fmt.Sprintf("Enabled CA changed from %s to %s", caName, mesh.GetEnabledCertificateAuthorityBackend().Name), nil
+	}
+	// generate snapshot if cert expired
+	generationUnixNano, err := strconv.Atoi(parts[0])
+	if err != nil {
+		return false, "", errors.Wrap(err, `invalid snapshot version format. Format should be "UnixNano;NameOfTheCA"`)
+	}
+	expiration := issuer.DefaultWorkloadCertValidityPeriod
+	if mesh.GetEnabledCertificateAuthorityBackend().GetDpCert().GetRotation().GetExpiration() != nil {
```

Comment:
I think makes sense to specify `MinWorkloadCertValidityPeriod` to avoid a mess with small values

Reply:
Can you elaborate here? What will be the mess with small values?